connector_openproject/components/importer.py

```diff
@@ -0,0 +1,408 @@
+            "mail_create_nolog": True,
+            "mail_track_log_only": True,
+            "mail_notrack": True,
+            "tracking_disable": True,
+        }
+
+    def import_dependencies(self, record):
+        self._import_link_dependency(record, OP_PROJECT_LINK)
+        self._import_link_dependency(record, OP_ASSIGNEE_LINK, raise_null=False)
+        self._import_link_dependency(record, OP_STATUS_LINK)
+
+    def _after_import(self, binding, record, for_create=False):
+        project_id_, project = self._link_to_internal(record, OP_PROJECT_LINK)
+        if not project.sync_activities == ACTIVITY_SYNC_NONE:
+            # Import work package activities in bulk (don't split into separate
+            # jobs).
+            self.component(
+                model_name="openproject.mail.message", usage="batch.importer"
+            ).run(binding.openproject_id, job_options={"delay": False})
+
+
+class OpenProjectAccountAnalyticLineImporter(Component):
+    _name = "openproject.account.analytic.line.importer"
+    _inherit = "base.openproject.importer"
+    _apply_on = "openproject.account.analytic.line"
+
+    def import_dependencies(self, record):
+        self._import_link_dependency(record, OP_WORK_PACKAGE_LINK, raise_null=False)
+        self._import_link_dependency(record, OP_USER_LINK)
+
+
+class OpenProjectMailMessageImporter(Component):
+    _name = "openproject.mail.message.importer"
+    _inherit = "base.openproject.importer"
+    _apply_on = "openproject.mail.message"
+
+    def _should_skip(self, record):
+        wp_id_, wp = self._link_to_internal(record, OP_WORK_PACKAGE_LINK)
+        project = self.env["openproject.project.project"].search(
+            [
+                ("odoo_id", "=", wp.project_id.id),
+            ]
+        )
+        enabled_for = project.sync_activities
+        activity_type = record["_type"]
+        if should_skip_activity(activity_type, enabled_for):
+            return f"Skipping sync for activity type: {activity_type}"
+
+    def import_dependencies(self, record):
+        self._import_link_dependency(record, OP_WORK_PACKAGE_LINK)
+        self._import_link_dependency(record, OP_USER_LINK)
+
+
+class ImageImporter(AbstractComponent):
+    _name = "base.image.importer"
+    _inherit = [
+        "base.importer",
+    ]
+    _usage = "image.importer"
+
+    def _download_image(self, url, timeout=None):
+        headers = {
+            "User-Agent": USER_AGENT,
+        }
+
+        try:
+            response = requests.get(url, headers=headers, timeout=timeout)
+        except requests.exceptions.Timeout:
+            raise NetworkRetryableError("Timeout while downloading image")
+        return response.content if response.ok else None
+
+    def run(self, url, binding_model, record_id, image_field, timeout=None):
+        image = self._download_image(url, timeout=timeout)
+        if not image:
+            return "No image could be downloaded"
```

Comment:
Not necessarily fail it, but at least we can return some result that informs you whether it failed or not.

Like let say, instead of just string, we would return tuple (bool, str), where `bool` is value indicating whether import was success and another item in tuple, showing some message, like you do now.

When we return just message, its not very good indicator whether import succeeded or not.

If we would let say want to extend this:

with `res = super().run()`, we might want to know result of import easily. And it would be hard to find it out by just message.
So something like

```python
   success, msg = super().run()
   if success:
       # do something
```